### PR TITLE
Feature/angular usage notes

### PR DESCRIPTION
**Pull request by nalbion**
Are the examples in the most recent commit valid?

### CODE REVIEW

File: templates/angular-class.mustache [hunk 1]
```diff
@@ -1,5 +1,26 @@
 /*jshint -W069 */
 /*global angular:false */
+/**
+ * Implemented as a Factory Factory so that a base URL, cache & token can be configured
+ * differently for testing and production.
+ * usage:<pre>
+angular.module('demo', ['{{&moduleName}}'])
```
Comment (wcandillon) [inline templates/angular-class.mustache]: @nalbion shouldn't these be prefixed by *?

**Comment by nalbion** [comment]
Can you explain to me how $on and $broadcast are used?  Would be good to have some examples

**Comment by nalbion** [comment]
Also, you don't seem to be using `this.cache` anywhere - it has to be passed in with each method's parameters

**Comment by wcandillon** [comment]
I think the cache thing can be removed indeed.
`$on` and `$broadcast` were here so people could subscribe to changes on a resource but I guess it could be removed too.

**Comment by wcandillon** [comment]
@nalbion Would it be possible to clean up the readme into multiple sections? it's starts to get overloaded now.

**Comment by nalbion** [comment]
I've done a little refactoring in another branch - I'll probably leave it there for now though.
https://github.com/nalbion/swagger-js-codegen/tree/feature/angularjs-closure

**Comment by wcandillon** [comment]
@nalbion feel free to reopen if necessary.

**Pull request closed without merging**